_posts/2020-10-30-snowflake-dataOps.md

```diff
@@ -0,0 +1,165 @@
+
+Last step is allowing code build in your pipeline to access the key in secrets manager.
+To do that find out the arn of your secrets manager:
+<center><img src="/img/blog/dataops/secretsmanager-3.png" /></center><br/>
+Open aws_seed.yaml file and update line number 404 with the arn to your secret manager password
+```
+  - <The arn to the secrets manager that holds Snowflake Password>
+```
+
+# Step 3: Creating Snowflake resources
+Now it is time to create Snowflake database, roles, grants and other resources that pipeline require in order to implement changes in snowflake.
+<i>Run below commands on Snowflake worksheet or execute them using snowsql</i>
+```
+CREATE DATABASE pipeline_db_migration_plan;
+
+CREATE ROLE pipeline_role;
+GRANT ROLE pipeline_role TO USER pipeline_sys_user;
+GRANT ALL ON DATABASE pipeline_db_migration_plan TO ROLE pipeline_role;
+GRANT ALL ON SCHEMA public TO ROLE pipeline_role;
+CREATE WAREHOUSE pipeline_warehouse;
+GRANT USAGE ON WAREHOUSE pipeline_warehouse TO ROLE pipeline_role;
+
+// Ideally you only want to grant permissions that your pipeline requires. Granting SYSADMIN is not encouraged
+grant role SYSADMIN to role pipeline_role;
+
+```
+
+Optional:
+In my sql scripts, I am creating tables in my_db database and my_schema schema. So if you want my template scripts run seccessfully on your snowflake account, you will need to run below statements too 
+```
+create database my_db;
+grant all on database my_db to role pipeline_role;
+
+create schema my_schema;
+grant all on schema my_schema to role pipeline_role;
+```
+
+# Step 4: Update pipeline parameters 
+Last step before creating your pipeline is to update parameters with the naming conventions you used to create your snowflake resources. If you have followed my naming convention you probably will not need to change anything except SnowflakeAccount.
+Update both parameter files pipeline/aws_seed-cli-parameters.json and aws_seed.json to match resources you created in snowflake
+
+```
+ "SnowflakeAccount": "<ACCOUNTNAME>.<REGION>",
+ "SnowflakeUsername": "pipeline_sys_user",
+ "SnowflakeMigrationDatabaseName": "pipeline_db_migration_plan",
+ "SnowflakeWarehouse": "pipeline_warehouse",
+ "SnowflakeRole": "pipeline_role",
+```
+
+```
+  {
+    "ParameterKey": "SnowflakeAccount",
+    "ParameterValue": "<ACCOUNTNAME>.<REGION>"
+  },
+  {
+    "ParameterKey": "SnowflakeUsername",
+    "ParameterValue": "pipeline_sys_user"
+  },
+  {
+    "ParameterKey": "SnowflakeWarehouse",
+    "ParameterValue": "pipeline_warehouse"
+  },
+  {
+    "ParameterKey": "SnowflakeMigrationDatabaseName",
+    "ParameterValue": "pipeline_db_migration_plan"
+  },
+  {
+    "ParameterKey": "SnowflakeRole",
+    "ParameterValue": "pipeline_role"
+  }
+```
+
+# Step 4: Deploy your code to AWS
+To deploy the infrastructure for your pipeline, you will need to first setup your aws credentials in your terminal. Next you will need to execute init.sh file
+
+```
+sh pipeline/init.sh
```

Comment:
could be worth noting that the aws user you are running as will need admin-like privileges, e.g. be able to create iam roles etc.